src/UserService.Business/Commands/Certificate/CreateCertificateCommand.cs

```diff
@@ -2,129 +2,118 @@
-            }
-
-            Guid? imageId = GetImageId(request.Image, errors);
-
-            if (!imageId.HasValue)
-            {
-                return new OperationResultResponse<Guid>
-                {
-                    Status = OperationResultStatusType.Failed,
-                    Errors = errors
-                };
-            }
-
-            var dbUserCertificate = _mapper.Map(request, imageId.Value);
-
-            _certificateRepository.Add(dbUserCertificate);
-
-            return new OperationResultResponse<Guid>
-            {
-                Status = OperationResultStatusType.FullSuccess,
-                Body = dbUserCertificate.Id
-            };
+          return responsedMsg.Body.ImagesIds;
         }
+
+        _logger.LogWarning(logMessage, userId, string.Join(',', responsedMsg.Errors));
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc, logMessage, userId, exc.Message);
+      }
+
+      errors.Add($"Can not add certificate image for user {userId}. Please try again later.");
+
+      return null;
+    }
+
+    public CreateCertificateCommand(
+    IAccessValidator accessValidator,
+    IHttpContextAccessor httpContextAccessor,
+    IDbUserCertificateMapper mapper,
+    IUserRepository userRepository,
+    ICertificateRepository certificateRepository,
+    IRequestClient<ICreateImagesRequest> rcAddIImage,
+    ILogger<CreateCertificateCommand> logger)
+    {
+      _accessValidator = accessValidator;
+      _httpContextAccessor = httpContextAccessor;
+      _mapper = mapper;
+      _userRepository = userRepository;
+      _certificateRepository = certificateRepository;
+      _rcImage = rcAddIImage;
+      _logger = logger;
+    }
+
+    public async Task<OperationResultResponse<Guid>> Execute(CreateCertificateRequest request)
+    {
+      OperationResultResponse<Guid> response = new();
+      Guid senderId = _httpContextAccessor.HttpContext.GetUserId();
+
+      if (!_accessValidator.HasRights(Rights.AddEditRemoveUsers) && senderId != request.UserId)
+      {
+        response.Errors.Add("Not enough rights");
+
+        response.Status = OperationResultStatusType.Failed;
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Forbidden;
+
+        return response;
+      }
+
+      List<AddImageRequest> requestImages = request.Images;
+      List<Guid> createdImagesIds = requestImages is not null && requestImages.Any()? 
+        await CreateImages(request, senderId, response.Errors) : new();
+      DbUserCertificate dbUserCertificate = _mapper.Map(request, createdImagesIds);
+
+      _certificateRepository.Add(dbUserCertificate);
+
+      response.Body = dbUserCertificate.Id;
+      response.Status = response.Errors.Any() ?
+        OperationResultStatusType.PartialSuccess : OperationResultStatusType.FullSuccess;
+      _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.OK;
```

Comment:
how i know you can don't write this, because OK is default value

Reply:
you are right but i think we need write it to to clearly indicate our intentions. if others dont agree with that - i`ll remove it

Reply:
you create entity - use HttpStatusCode.Created

Reply:
++